### PR TITLE
Status code reworked, woraround for #29, improve debug logging

### PR DESCRIPTION
**Pull request by notandy**

### CODE REVIEW

File: octavia_f5/controller/worker/f5agent_driver.py [hunk 1]
```diff
@@ -121,3 +127,18 @@ def tenant_delete(bigip, tenant):
...
     return bigip.delete(tenants=[tenant])
+
+
+def member_create(bigip, member):
+    """Patches new member into existing pool
+
+    :param bigip: bigip instance
+    :param member: octavia member object
+    """
+    path = '{}/{}/{}/members/-'.format(
```
Comment (laerlingSAP) [inline octavia_f5/controller/worker/f5agent_driver.py]: Is the hyphen intended?

File: octavia_f5/restclient/as3objects/pool.py [hunk 2]
```diff
@@ -73,6 +78,8 @@ def get_pool(pool):
                     member_hm.set_target_port(member.monitor_port)
                 name = m_member.get_name(member.id)
                 entities.append((name, member_hm))
+                if not utils.pending_delete(member):
+                    service_args['monitors'].append(Pointer(use=name))
```
Comment (laerlingSAP) [inline octavia_f5/restclient/as3objects/pool.py]: Please add comment hinting to this being part of the workaround, similar to line 72.

File: octavia_f5/controller/worker/status.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        """Set provisioning_state of octavia object and all ancestors to
+        ACTIVE.
+
+        :param obj: octavia object like loadbalancer, pools, etc.
+        """
+        self._set_obj_and_ancestors(obj)
+
+    def set_deleted(self, obj):
+        """Set provisioning_state of octavia object to DELETED and all
+        ancestors to ACTIVE.
```
Comment (laerlingSAP) [inline octavia_f5/controller/worker/status.py]: Typo: ACTIVE instead of DELETED

File: octavia_f5/controller/worker/status.py [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+        :param obj: octavia object like loadbalancer, pools, etc.
+        """
+        self._set_obj_and_ancestors(obj, lib_consts.DELETED)
+
+    def set_error(self, obj):
+        """Set provisioning_state of octavia object to ERROR and all
+        ancestors to ACTIVE.
```
Comment (laerlingSAP) [inline octavia_f5/controller/worker/status.py]: Typo: ACTIVE instead of ERROR

File: octavia_f5/controller/worker/status.py [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+        :param obj: octavia object like loadbalancer, pools, etc.
+        """
+        self._set_obj_and_ancestors(obj, lib_consts.ERROR)
+
+    def update_status(self, loadbalancers):
+        """Set provisioning_state of loadbalancers and all it's
+        childs to ACTIVE if PENDING_UPDATE or PENDING_CREATE, else
```
Comment (laerlingSAP) [inline octavia_f5/controller/worker/status.py]: Typo: childs instead of children

File: octavia_f5/controller/worker/status.py [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+                lib_consts.L7RULES: [obj_status],
+                lib_consts.L7POLICIES: [self._status_obj(obj.l7policy)],
+                lib_consts.LISTENERS: [self._status_obj(obj.l7policy.listener)],
+                lib_consts.LOADBALANCERS: [self._status_obj(
+                    obj.l7policy.listener.load_balancer)]
+            })
+
+    @staticmethod
+    def _status_obj(obj,
+                    provisioning_status=lib_consts.ACTIVE):
```
Comment (laerlingSAP) [inline octavia_f5/controller/worker/status.py]: Please add short docstring

File: octavia_f5/controller/worker/status.py [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+        childs to ACTIVE if PENDING_UPDATE or PENDING_CREATE, else
+        DELETED for PENDING_DELETED.
+
+        :param loadbalancers: octavia loadbalancers list
+        """
+        for loadbalancer in loadbalancers:
+            if utils.pending_delete(loadbalancer):
+                self.set_deleted(loadbalancer)
+            else:
+                self.set_active(loadbalancer)
```
Comment (laerlingSAP) [inline octavia_f5/controller/worker/status.py]: This code section is used multiple times. Maybe separate it into a function?
E. g. `set_active_or_deleted`

File: octavia_f5/controller/worker/f5agent_driver.py (hunk 1 above)
Comment (notandy) [reply octavia_f5/controller/worker/f5agent_driver.py]: yes

File: octavia_f5/restclient/as3objects/pool.py (hunk 2 above)
Comment (notandy) [reply octavia_f5/restclient/as3objects/pool.py]: this is not part of the workaround, but of the custom member address/port. just a fix

File: octavia_f5/controller/worker/status.py (hunk 3 above)
Comment (laerlingSAP) [reply octavia_f5/controller/worker/status.py]: Feedback from @notandy: This is intended. Deletion is not recursive. Child objects are set to PENDING_UPDATE though, that's why we set them back to ACTIVE.

File: octavia_f5/controller/worker/status.py (hunk 4 above)
Comment (laerlingSAP) [reply octavia_f5/controller/worker/status.py]: Same reason as for line 47.

**Pull request merged**